### PR TITLE
[arp_update]: Flush neighbors with incorrect MAC info

### PR DESCRIPTION
**Pull request by theasianpianist**
#### Why I did it
- Prevent packet drops due to MAC expiration in hardware
- Mitigate MAC mismatch between kernel and APPL_DB neighbor table

##### Work item tracking
- Microsoft ADO **25135974** and **25647679**:

#### How to verify it
- Load script onto some DUT, wait for MAC entries to expire (`show arp` will have entries with 'Iface' listed as '-'). Run the arp_update script. After several seconds confirm that the expired entries are no longer present in `show arp`
- For some learned neighbor, manually set the APPL_DB MAC address in the neighbor table to a value different from the MAC address learned in the kernel. Run the arp_update script, confirm that the neighbor is flushed.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ 20220531.38] 
- [ ]

### CODE REVIEW

File: files/scripts/arp_update [hunk 1]
```diff
@@ -70,6 +75,29 @@ while /bin/true; do
...
+  # Flush neighbor entries with MAC mismatch between kernel and APPL_DB
+  KERNEL_NEIGH=$(ip neigh show | grep -v "fe80" | grep -v "FAILED\|INCOMPLETE" | cut -d ' ' -f 1,3,5 --output-delimiter=',' | tr -d ' ')
+  for neigh in $KERNEL_NEIGH; do
+        ip="$( cut -d ',' -f 1 <<< "$neigh" )"
+        intf="$( cut -d ',' -f 2 <<< "$neigh" )"
+        kernel_mac="$( cut -d ',' -f 3 <<< "$neigh" )"
+        appl_db_mac="$(sonic-db-cli APPL_DB hget NEIGH_TABLE:$intf:$ip neigh)"
+        if [[ $kernel_mac != $appl_db_mac ]]; then
+            logger -p warning "MAC mismatch for ${ip} on ${intf} - kernel: ${kernel_mac}, APPL_DB: ${appl_db_mac}"
+            ip neigh flush $ip
```
Comment (prsunny) [inline files/scripts/arp_update:97]: for the mistmatched ones, we would want want to flush and also ping

**Review by prsunny — APPROVED** [review]

**Comment by StormLiangMS** [comment]
Hi @theasianpianist @prsunny  could you run test against with 202305 branch?

**Comment by theasianpianist** [comment]
> Hi @theasianpianist @prsunny could you run test against 202305 branch to avoid regression at this late stage.

confirmed the functionality is working on 202305 image

**Comment by StormLiangMS** [comment]
@theasianpianist @prsunny do we need this for 202311 also?

**Comment by prsunny** [comment]
> @theasianpianist @prsunny do we need this for 202311 also?

isn't this already in 202311 as the PR was merged on 30 Nov'23. If not, yes, we need this

**Comment by theasianpianist** [comment]
> @theasianpianist @prsunny do we need this for 202311 also?

looks like it's already included https://github.com/sonic-net/sonic-buildimage/commit/7d308e340a0c847f39cd203ad4a2c287b9c68e34

**Pull request merged**